### DISCUSSION TITLE
Potential for a Windows App Target?

### DISCUSSION THREAD

**Discussion by fbartho** [skiptools/skip#199] [Ideas]
TheBrowserCompany (makers of the Arc Browser - https://arc.net) released their stack for Swift and SwiftUI bindings on Windows.
- https://github.com/thebrowsercompany/swift-winrt
- https://github.com/thebrowsercompany/swift-winui

Have you investigated/considered using that to make a Windows target? (I did search for other discussions, and found this one: https://github.com/orgs/skiptools/discussions/163#discussioncomment-9919141 with a glorious reply from @marcprux — however the thrust of the discussion was pretty different)

I’m currently developing a iOS/iPadOS/macOS application, and could see us deploying that to Android using Skip. It would be pretty compelling if Skip could also help customers assemble their Windows app.

**Comment by marcprux** [comment]
The `swift-winui` project is indeed very interesting. While we (Skip) are focused solely on iOS and Android app development, there's technically no reason why your project couldn't *also* work with TBC's Windows port as well. Since we don't intrude on the pure-Swift side of your project, you are free to support any additional build destinations and runtime environments that can be used with Swift. For a simple example, you can also target macOS, tvOS, and watchOS in your Skip project, you would just likely disable the transpiler plugin, since it wouldn't be useful in that context.

If you investigate this, please let us know how it goes. I would be great to have an example of a vastly cross-platform projects that is built with a combination of Skip and other technologies, like `swift-winrt` or the recent [Gnome project](https://www.swift.org/blog/adwaita-swift/) for Linux.

**Comment by clarkezone** [comment]
I'm interested in this as well @fbartho did you do any experimentation?  Looks like the swift-winui project is sunset.  What I'm looking for is a mapping layer from swiftui onto something platform native for windows without any Kotlin or JVM dependencies. could be WinUI or even something layered on top of low-level building blogs like DirectWrite / Direct3D / win::ui::composition.  Another idea would be to use something like Rive.App with swift bindings.

**Reply by fbartho** [reply]
No, I did not explore much further. I checked on the space a few times, and had a feeling that none of the options available had enough maturity or commitment behind them for me to trust their longevity, and I don’t have the bandwidth to own my own solution (or fork one of the existing if it went under).

Similarly, I’m a fan conceptually of skiptools, but as a single-vendor commercial solution, I have the same fear/risk. If skiptools produced a WASM build too, that would be of higher value to me than the Windows target.
